core/src/main/java/com/netflix/conductor/service/ExecutionService.java

```diff
@@ -280,33 +280,29 @@ private boolean requeue(Task pending) throws Exception {
 		return queue.pushIfNotExists(QueueUtils.getQueueName(pending), pending.getTaskId(), callback);
 	}
 
-	public List<Workflow> getWorkflowInstances(String workflowName, String correlationId, boolean includeClosed, boolean includeTasks)
-			throws Exception {
-		List<Workflow> workflows = executor.getStatusByCorrelationId(workflowName, correlationId, includeClosed);
-		if (includeTasks) {
-			workflows.forEach(wf -> {
-				List<Task> tasks;
-				try {
-					tasks = edao.getTasksForWorkflow(wf.getWorkflowId());
-				} catch (Exception e) {
-					throw new RuntimeException(e);
-				}
-				wf.setTasks(tasks);
-			});
+	public List<Workflow> getWorkflowInstances(String workflowName, String correlationId, boolean includeClosed, boolean includeTasks) {
+
+		List<Workflow> workflows = executionDAO.getWorkflowsByCorrelationId(correlationId, includeTasks);
```

Comment:
This is duplicated with WorkflowExecutor.getStatusByCorrelationId after another change above, why not just call it here?